### PR TITLE
Feature/persistent chat

### PR DESCRIPTION
**Pull request by mehul1011**
Changes for #4 issue,
I have implemented the persistent storage logic, for UUID I am thinking about what would be a better way since different chats Selected would require different UUIDs and against that fetching related chat data from store/localStorage

For this issue, I have used NgRx store (basically this would be a source of truth for fetching messages until the browser tab is closed where localStorage comes in to again load into the store the relevant data)

I have used these since I am using them all for the first time, let me know if you see any improvement.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,18 +14,23 @@
     "@angular/cdk": "^13.3.9",
     "@angular/common": "^14.3.0",
     "@angular/compiler": "^14.3.0",
-    "@angular/core": "^14.3.0",
+    "@angular/core": "^14.0.0",
```
Comment (yagizhanNY) [inline package.json:17]: Is there any problem with version 14.3.0?

File: src/app/services/chat.service.ts [hunk 2]
```diff
@@ -25,7 +25,7 @@ export class ChatService {
       {
         headers: {
           'Content-Type': 'application/json',
-          'User-Agent': 'OpenAPI-Generator/1.0/Javascript',
+          'X-User-Agent': 'OpenAPI-Generator/1.0/Javascript',
```
Comment (yagizhanNY) [inline src/app/services/chat.service.ts:28]: Is it necessary?

File: package.json (hunk 1 above)
Comment (mehul1011) [reply package.json:17]: NgRx store was giving issues while installing, basically i had to find a compatible version with angular core.

It was some error "root has 14.3.0 

But requirement is not meeting from NgRx store version."

I can replicate that version if you need to see.

File: src/app/services/chat.service.ts (hunk 2 above)
Comment (mehul1011) [reply src/app/services/chat.service.ts:28]: Not necessarily, but in console it was giving an error for unsafe usage on headers.

File: package.json (hunk 1 above)
Comment (yagizhanNY) [reply package.json:17]: no, it can stay. I just want to know :)

Comment (mehul1011) [reply package.json:17]: > no, it can stay. I just want to know :)

Sure i will replicate this and show it to you with a screenshot.

Comment (mehul1011) [reply package.json:17]: openai-chatgpt3-clone (feature/persistentChat) $ npm install
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: open-ai-web@0.0.0
npm ERR! Found: @angular/core@14.3.0
npm ERR! node_modules/@angular/core
npm ERR!   @angular/core@"^14.3.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/core@"^15.0.0" from @ngrx/store@15.4.0
npm ERR! node_modules/@ngrx/store
npm ERR!   @ngrx/store@"^15.4.0" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /Users/../.npm/_logs/2023-04-14T10_06_55_570Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/../.npm/_logs/2023-04-14T10_06_55_570Z-debug-0.log

Error was something like this, i changed things from before when they were coming, seems like i was trying to install latest of NgRx/Store but it was not compatible with angular/core version, i read they have to point to same version number like 15.x.x == 15.x.x, correct me if i am wrong.

**Pull request merged**